src/vulkan/buffer_descriptor.cc

```diff
@@ -47,12 +47,23 @@ Result BufferDescriptor::CreateResourceIfNeeded() {
   uint32_t size_in_bytes =
       amber_buffer ? static_cast<uint32_t>(amber_buffer->ValuePtr()->size())
                    : 0;
-  transfer_buffer_ = MakeUnique<TransferBuffer>(device_, size_in_bytes);
+  transfer_buffer_ = MakeUnique<TransferBuffer>(
+      device_, size_in_bytes,
+      amber_buffer ? amber_buffer->GetFormat() : nullptr);
+  VkBufferUsageFlags flags =
+      VK_BUFFER_USAGE_TRANSFER_SRC_BIT | VK_BUFFER_USAGE_TRANSFER_DST_BIT;
+  if (IsUniformBuffer()) {
+    flags |= VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT;
+  } else if (IsStorageBuffer()) {
+    flags |= VK_BUFFER_USAGE_STORAGE_BUFFER_BIT;
+  } else if (IsUniformTexelBuffer()) {
+    flags |= VK_BUFFER_USAGE_UNIFORM_TEXEL_BUFFER_BIT;
+  } else {
+    assert(IsStorageTexelBuffer());
```

Comment:
Instead of `assert` this should return a failed Result.

Reply:
done